mne/event.py

```diff
@@ -827,3 +828,289 @@ def concatenate_events(events, first_samps, last_samps):
+        events compatible with the transition, bitwise ORed together:
+        e.g. [t1, 0, 5] means that events 1 and 3 occurred at time t1,
+        as 2**(1 - 1) + 2**(3 - 1) = 5. """
+        events_ = mne_events.copy()
+        events_[:, 1] = 0
+        events_[:, 2] = 0
+        for n, ev in self._events.iteritems():
+            if ev.in_use:
+                pre_ok = (
+                    np.bitwise_and(ev.oldmask, mne_events[:, 1]) == ev.oldbits)
+                post_ok = (
+                    np.bitwise_and(ev.newbits, mne_events[:, 2]) == ev.newbits)
+                ok_ind = np.where(pre_ok & post_ok)
+                events_[ok_ind, 2] |= 1 << (n - 1)
+        return events_
+
+    def _mne_events_to_category_t0(self, cat, mne_events, sfreq):
+        """ Translate mne_events to reference times (t0) for epochs in a given
+        DACQ averaging category cat. """
+        events = self._events_mne_to_dacq(mne_events)
+        times = events[:, 0]
+        # indices of times where ref. event occurs
+        refEvents_inds = np.where(events[:, 2] & (1 << cat.event - 1))[0]
+        refEvents_t = times[refEvents_inds]
+        if cat.reqevent:
+            # indices of times where req. event occurs
+            reqEvents_inds = np.where(events[:, 2] & (
+                1 << cat.reqevent - 1))[0]
+            reqEvents_t = times[reqEvents_inds]
+            # relative (to refevent) time window (in samples) where req. event
+            # must occur (e.g. [0 200])
+            win = np.round(np.array(
+                sorted([0, (-1)**(cat.reqwhen) * cat.reqwithin])) * sfreq)
+            refEvents_wins = refEvents_t[:, None] + win
+            req_acc = np.zeros(refEvents_inds.shape, dtype=bool)
+            for t in reqEvents_t:
+                # mark time windows where req. condition is satisfied
+                reqEvent_in_win = np.logical_and(
+                    t >= refEvents_wins[:, 0], t <= refEvents_wins[:, 1])
+                req_acc |= reqEvent_in_win
+            # drop ref. events where req. event condition is not satisfied
+            refEvents_inds = refEvents_inds[np.where(req_acc)]
+            refEvents_t = times[refEvents_inds]
+        # adjust for trigger-stimulus delay by delaying the ref. event
+        refEvents_t += int(np.round(self._events[cat.event].delay * sfreq))
+        return refEvents_t
+
+    @property
+    def categories(self):
+        """ Return list of categories in DACQ defined order. Only returns
+        categories marked active in DACQ. """
+        return sorted(self._categories_in_use,
+                      key=lambda cat: cat.index)
+
+    @property
+    def events(self):
+        """ Return list of events in DACQ order. Only returns events that
+        are referred to by a DACQ category. """
+        return sorted(self._events_in_use,
+                      key=lambda ev: ev.index)
+
+    def get_mne_rejection_dict(self):
```

Comment:
I would create the reject dict in the init and just have a .reject attribute. This would remove one method.